File: wf-api-integration/references/beneficiary/bind/java/model/response/BindBeneficiaryResponse.java
package {basePackage}.wf.model.response;

import {basePackage}.wf.model.domain.Beneficiary;

import org.apache.commons.lang3.builder.ToStringBuilder;
import org.apache.commons.lang3.builder.ToStringStyle;

/**
 * WorldFirst bindBeneficiary 响应对象。
 *
 */
public class BindBeneficiaryResponse {

    /** API 调用结果 */
    private Result result;

    /** 绑定的收款人信息 */
    private Beneficiary beneficiary;

    public Result getResult() {
        return result;
    }

    public void setResult(Result result) {
        this.result = result;
    }

    public Beneficiary getBeneficiary() {
        return beneficiary;
    }

    public void setBeneficiary(Beneficiary beneficiary) {
        this.beneficiary = beneficiary;
    }

    public boolean isSuccess() {
        return result != null && "S".equals(result.getResultStatus());
    }

    @Override
    public String toString() {
        return ToStringBuilder.reflectionToString(this, ToStringStyle.SHORT_PREFIX_STYLE);
    }
}

File: wf-api-integration/references/beneficiary/bind/golang/model/response/bind_beneficiary_response.go
package response

import (
	"{moduleName}/wf/model/domain"
)

// Result represents the standard WF API result object.
type Result struct {
	ResultStatus  string `json:"resultStatus"`
	ResultCode    string `json:"resultCode"`
	ResultMessage string `json:"resultMessage"`
}

// BindBeneficiaryResponse is the response for bindBeneficiary API.
type BindBeneficiaryResponse struct {
	Result      Result              `json:"result"`
	Beneficiary *domain.Beneficiary `json:"beneficiary,omitempty"`
}

